File: src/main/java/com/saiph/application/GestionRH/Domain/dto/EventDto.java
package com.saiph.application.GestionRH.Domain.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.saiph.application.GestionRH.Domain.entities.User;
import jakarta.persistence.ManyToMany;
import lombok.Getter;
import lombok.NonNull;
import lombok.Setter;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;


@Getter
@Setter
public class EventDto extends GenericDto {


    @NonNull
        @JsonFormat(pattern = "yyyy-MM-dd")
    private Date dateD;
    @NonNull
        @JsonFormat(pattern = "yyyy-MM-dd")
    private Date dateF;
    @NonNull
    private Integer duree;
    @NonNull
    private String lieu;
    @NonNull
    private Integer nbrPlace;
    @NonNull
    private String img;
    @NonNull
    private String titre;
    @NonNull
    private String description;



    @ManyToMany
    private List<User> ListEmploye = new ArrayList<>();
}

File: Gestion-Rh-UI/src/app/services/models/event-dto.ts
/* tslint:disable */
/* eslint-disable */
import { Utilisateur } from '../models/utilisateur';
export interface EventDto {
  createdBy?: string;
  createdDate?: string;
  dateD?: string;
  dateF?: string;
  description?: string;
  duree?: number;
  id?: number;
  img?: string;
  lastModifiedBy?: string;
  lastModifiedDate?: string;
  lieu?: string;
  listEmploye?: Array<Utilisateur>;
  nbrPlace?: number;
  titre?: string;
}
